pkg/ottl/parser.go

```diff
@@ -44,13 +55,21 @@ type Statement[K any] struct {
 func (s *Statement[K]) Execute(ctx context.Context, tCtx K) (any, bool, error) {
 	condition, err := s.condition.Eval(ctx, tCtx)
 	if err != nil {
-		return nil, false, err
+		if s.errorMode == PropagateError {
+			return nil, false, err
+		}
+		s.telemetrySettings.Logger.Error("error executing condition", zap.Error(err))
```

Comment:
I would enhance the error with the step information

if err != nil {
  err = fmt.Errorf("failed to execute condition: %w", err)
}

Reply:
Good idea.